packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,217 @@
+import { IResource, Resource, Tag } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { AcceptLanguage } from '.';
+import { IPortfolio } from './portfolio';
+import { AssociationManager } from './private/association-manager';
+import { CfnCloudFormationProduct } from './servicecatalog.generated';
+
+
+/**
+ * A Service Catalog product, currently only supports type CloudFormationProduct
+*/
+export interface IProduct extends IResource {
+
+  /**
+   * The id of the product
+   *
+   * @attribute
+   */
+  readonly id: string;
+
+  /**
+   * The name of the product
+   *
+   * @attribute
+   */
+  readonly name: string;
+
+  /**
+   * Add product to a portfolio
+   */
+  addToPortfolio(portfolio: IPortfolio): void;
+}
+
+
+/**
+* Represents a Service Catalog CloudFormation Product.
+*/
+abstract class ProductBase extends Resource implements IProduct {
+
+  /**
+  * The logical id of the product
+  */
+  public abstract readonly id: string;
+
+  /**
+  * The name of the product
+  */
+  public abstract readonly name: string;
+
+  /**
+   * Add product to a portfolio
+   */
+  addToPortfolio(portfolio: IPortfolio) {
+    AssociationManager.associateProductWithPortfolio(this, portfolio, this);
+  }
+
+}
+
+/**
+ * Properties of provisioning artifacts
+ */
+export interface ProvisioningArtifactProperties {
+
+
+  /**
+     * The description of the provisioning artifact
+     * @default
+     */
+  readonly description?: string;
+
+  /**
+     * Disable template validation.
+     * @default True
+     */
+  readonly disableTemplateValidation?: boolean;
+
+  /**
+     * The S3 url that points to the provisioning artifacts template
+     */
+  readonly templateUrl: string;
+
+  /**
+   * The name of the provisioning artifact.
+     * @default
```

Comment:
What's happening here? Misindented code, no default description...

Reply:
I think we missed a lot of these assuming they would be caught by linting, will look into making sure it gets picked up